Automattic/jetpack
# Allow plugins that consume the Sync package to configure the data settings

#### Changes proposed in this Pull Request:
* Allow plugins that consume the sync package to configure the data settings

#### Jetpack product discussion
* p9dueE-2ZE-p2

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
* Get a test site with any Jetpack plugin that uses Sync (Jetpack, Backup).
* Connect site, observe Sync activity - see the full sync and all added data being synced correctly.
* Change the settings to minimal using the Config object:

```php

    use Automattic\Jetpack\Sync\Data_Settings;

    // ....
    // Change the existing `ensure` call to this: 
    $config->ensure( 'sync', Data_Settings::MUST_SYNC_DATA_SETTINGS );
```

* Observe that on reconnection a full sync no longer happens, data doesn't get synced and you can only see callables getting synced.
* Try again with multiple plugins installed, see that the settings get combined, not overwritten.
